lib/zoneddatetime.ts

```diff
@@ -212,9 +203,29 @@ export class ZonedDateTime implements Temporal.ZonedDateTime {
     if (!props) {
       throw new TypeError('invalid zoned-date-time-like');
     }
-    let fields = ES.ToTemporalZonedDateTimeFields(this, fieldNames) as any;
+    const entries = [
+      ['day', undefined],
+      ['hour', 0],
+      ['microsecond', 0],
+      ['millisecond', 0],
+      ['minute', 0],
+      ['month', undefined],
+      ['monthCode', undefined],
+      ['nanosecond', 0],
+      ['second', 0],
+      ['year', undefined],
+      ['offset'],
+      ['timeZone']
+    ];
```

Comment:
In other types, the `entries` variable is typed (via a type annotation) as something like this:

```js
const entries: [keyof Temporal.ZonedDateTimeLike, 0 | undefined][]
```

This removes the need for the `as any` below.

usually these `entries` values are typed `as const` or with a type annotation on